addons/mail/static/src/components/notification_list/notification_list.js

```diff
@@ -133,9 +133,8 @@ class NotificationList extends Component {
         if (props.filter === 'all') {
             const notificationGroups = this.env.messaging.notificationGroupManager.groups;
             notifications = Object.values(notificationGroups)
-                .sort((group1, group2) =>
-                    group1.date.isAfter(group2.date) ? -1 : 1
-                ).map(notificationGroup => {
+                .sort((group1, group2) => group1.sequence > group2.sequence ? -1 : 1)
```

Comment:
That's not good. `sequence` must be consistent with most uses of this word in programming domain:

A (technical) number that defines order of current item in context of a list of items that also have a sequence (number). Lower sequence (number) comes before higher ones. Same sequence (number) have arbitrary order among themselves, as long as ordering on all combination of different sequence (numbers) are respected by first sentence.

I don't care whether this approach is motivated by ease of implementation (`Math.max` of current group or last message `id`)... well... actually I do mind that the motivation is too technical and therefore poorly justified. Please always choose concept-driven development over feasibly-driven development: it's much easier to define good concept and consequently make good code from it than from just picking a feasible approach.